File: leetcode/js/20.ValidParenthesis.js
class Solution{
  
    static isValid(s){
      var match={
        ')':'(',
        '}':'{',
        ']':'['
      }
      var stack=[]
      for(let i in s){
        if(s[i]==='('||s[i]==='{'||s[i]==='['){
          stack.push(s[i])
          continue
        }
        if(stack.length === 0 || match[s[i]] != stack.pop()){
          return false
        }
      }
        return stack.length === 0
    }
  }
  
  console.log(Solution.isValid('()()'))

File: leetcode/python/20.ValidParenthesis.py
class Solution():
  @staticmethod
  def isValid(s):
    match={
      ')':'(',
      '}':'{',
      ']':'['
    }
    stack = []
    for i in range(len(s)):
      if s[i] == '(' or s[i] == '{' or s[i] == '[':
        stack.append(s[i])
        continue
      if len(stack) == 0 or match[s[i]] != stack.pop():
        return False
    return len(stack)==0
print(Solution.isValid('{{()}}'))